components/pdf_preview.jsx

```diff
@@ -42,17 +42,30 @@ export default class PDFPreview extends React.PureComponent {
     }
 
     componentDidMount() {
-        this.updateStateFromProps(this.props);
+        this.getPdfDocument();
     }
 
-    UNSAFE_componentWillReceiveProps(nextProps) { // eslint-disable-line camelcase
-        if (this.props.fileUrl !== nextProps.fileUrl) {
-            this.updateStateFromProps(nextProps);
-            this.pdfPagesRendered = {};
+    static getDerivedStateFromProps(props, state) {
+        if (props.fileUrl !== state.prevFileUrl) {
+            return {
+                pdf: null,
+                pdfPages: {},
+                pdfPagesLoaded: {},
+                numPages: 0,
+                loading: true,
+                success: false,
+                prevFileUrl: props.fileUrl,
+            };
         }
+        return null;
     }
 
     componentDidUpdate() {
+        if (this.state.pdf === null) {
```

Comment:
Similar to my previous comment, could you add a check here against the previous value of `state.pdf`?

Reply:
Same reasoning from my comment above applies to this component as well.

Reply:
And here would also be `this.props.fileUrl !== prevProps.fileUrl` as well then